localstack/services/awslambda/provider.py

```diff
@@ -218,6 +219,38 @@ def __init__(self) -> None:
         self.layer_fetcher = None
         lambda_hooks.inject_layer_fetcher.run(self)
 
+    def accept_state_visitor(self, visitor: StateVisitor):
+        visitor.visit(lambda_stores)
+
+    def on_after_state_load(self):
+        # TODO: provisioned concurrency
+        for account_id, account_bundle in lambda_stores.items():
+            for region_name, state in account_bundle.items():
+                for fn in state.functions.values():
+                    # only start functions that don't have a version manager yet
+                    try:
+                        self.lambda_service.get_lambda_version_manager(
+                            fn.latest().id.qualified_arn()
+                        )
+                    except ValueError:
+                        for fn_version in fn.versions.values():
+                            # restore the "Pending" state for every function version and start it
+                            new_state = VersionState(
+                                state=State.Pending,
+                                code=StateReasonCode.Creating,
+                                reason="The function is being created.",
+                            )
+                            new_config = dataclasses.replace(fn_version.config, state=new_state)
+                            new_version = dataclasses.replace(fn_version, config=new_config)
+                            fn.versions[fn_version.id.qualifier] = new_version
+                            self.lambda_service.create_function_version(fn_version)
+
+                            # Restore event source listeners
+                            for esm in state.event_source_mappings.values():
+                                EventSourceListener.start_listeners_for_asf(
+                                    esm, self.lambda_service
+                                )
```

Comment:
It seems like this should be independent from the function? Maybe wrong intendation?